src/printer.rs

```diff
@@ -428,15 +428,33 @@ impl Printer {
         response: &mut Response,
         encoding: Option<&'static Encoding>,
         mime: Option<&str>,
+        preserve_encoding: bool,
     ) -> anyhow::Result<()> {
         let url = response.url().clone();
         let content_type =
             mime.map_or_else(|| get_content_type(response.headers()), ContentType::from);
         let encoding = encoding.or_else(|| get_charset(response));
-        let compression_type = get_compression_type(response.headers());
+        let (may_decode, compression_type) = if !preserve_encoding {
+            let compression_type = get_compression_type(response.headers());
+            (true, compression_type)
+        } else {
+            (false, None)
+        };
```

Comment:
I don't think that it is possible since the logic below assumes that the content has been decompressed before colorizing, formatting, or decoding. We wouldn't want that logic to run on potentially encoded data. The other branches call code like decode_stream to interpret UTF8 or other format types.